drivers/ethernet/eth_stm32_hal.c

```diff
@@ -256,15 +256,17 @@ static inline void setup_mac_filter(ETH_HandleTypeDef *heth)
 
 	/* disable multicast perfect filtering */
 	tmp &= ~(ETH_MULTICASTFRAMESFILTER_PERFECTHASHTABLE |
-#if !defined(CONFIG_ETH_STM32_MULTICAST_FILTER)
 		 ETH_MULTICASTFRAMESFILTER_HASHTABLE |
-#endif /* CONFIG_ETH_STM32_MULTICAST_FILTER */
-		 ETH_MULTICASTFRAMESFILTER_PERFECT);
+		 ETH_MULTICASTFRAMESFILTER_PERFECT |
+		 ETH_MULTICASTFRAMESFILTER_NONE);
```

Comment:
Nitpick:
IIUC, This fully resets the filtering bits configuration bits, leaving the state to 0, which matches "perfect filtering" definition.
If I'm correct, it would be nice to fix the comment as well, which is misleading.

Reply:
Oh yeah that comment isn't great, better now?

Reply:
From my understanding, it's more accurate now. Thanks